router/transformer/transformer.go

```diff
@@ -357,6 +376,22 @@ func (trans *handle) ProxyRequest(ctx context.Context, proxyReqParams *ProxyRequ
 		}
 	}
 
+	/*
+		respData will be in ProxyResponseV0 or ProxyResponseV1
+	*/
+	transportResponse := oauthv2.TransportResponse{} // response that we get from oauth-interceptor in postRoundTrip
+	if (*trans.oauthV2EnabledLoader).Load() {
+		_ = json.Unmarshal(respData, &transportResponse)
```

Comment:
what about err handling

Reply:
No need to handle it as upstream you can get a string response even before performing OAuth operations like Refresh Token, Auth Status Toggle. It's alright if the respData structure doesn't follow oauthv2.TransportResponse struct